protocol-adapter/src/main/java/org/eclipse/ditto/protocoladapter/HeaderTranslator.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/org/documents/epl-2.0/index.php
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial contribution
+ */
+package org.eclipse.ditto.protocoladapter;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import javax.annotation.concurrent.Immutable;
+
+import org.eclipse.ditto.model.base.headers.DittoHeaderDefinition;
+import org.eclipse.ditto.model.base.headers.DittoHeaders;
+import org.eclipse.ditto.model.base.headers.DittoHeadersBuilder;
+import org.eclipse.ditto.model.base.headers.HeaderDefinition;
+
+/**
+ * Utility for translating Headers from external sources or to external sources.
+ * <p>
+ * Does so by applying blacklisting based on {@link HeaderDefinition}s.
+ * </p>
+ */
+@Immutable
+public final class HeaderTranslator {
+
+    private final Map<String, HeaderDefinition> headerDefinitionMap;
+
+    private HeaderTranslator(final Map<String, HeaderDefinition> headerDefinitionMap) {
+        this.headerDefinitionMap = Collections.unmodifiableMap(headerDefinitionMap);
+    }
+
+    /**
+     * Construct a Ditto header translator that knows about all header definitions enumerated in
+     * {@link DittoHeaderDefinition}.
+     *
+     * @return the Ditto header translator.
+     */
+    public static HeaderTranslator of() {
+        return of(DittoHeaderDefinition.values());
+    }
+
+    /**
+     * Construct a Ditto header translator from arrays of header definitions.
+     *
+     * @param headerDefinitions arrays of header definitions.
+     * @return the Ditto header translator that knows about the given definitions.
+     */
+    public static HeaderTranslator of(final HeaderDefinition[]... headerDefinitions) {
+        return new HeaderTranslator(Arrays.stream(headerDefinitions)
+                .flatMap(Arrays::stream)
+                .collect(Collectors.toMap(HeaderDefinition::getKey, Function.identity())));
+    }
+
+    /**
+     * Read Ditto headers from external headers.
+     *
+     * @param externalHeaders external headers as a map.
+     * @return Ditto headers initialized with values from external headers.
+     */
+    public DittoHeaders fromExternalHeaders(final Map<String, String> externalHeaders) {
+        final DittoHeadersBuilder builder = DittoHeaders.newBuilder();
+        externalHeaders.forEach((key, value) -> {
+            final HeaderDefinition definition = headerDefinitionMap.get(key);
```

Comment:
Case sensitivity plays a role here - if a "Authorization" key is in the `externalHeaders`, a "authorization" in the `headerDefinitionMap` won't match

Reply:
We should also add a test for case sensitivity